src/instrument_meta.py

```diff
@@ -0,0 +1,307 @@
+            except Exception as err:
+                error_msg = 'Failed to insert instrument meta record -'\
+                    f' err: {err}'
+                print(f'Submit PUT instrument meta error: {error_msg}')
+                return self.failed_request(error_msg)
+
+    def put_instrument_meta(self):
+        session = stm.get_session()
+
+        insert_stmt = insert(im).values(
+            name = self.instrument_meta.name,
+            num_channels = self.instrument_meta.num_channels,
+            scan_angle = self.instrument_meta.scan_angle,
+            created_at = datetime.utcnow(),
+            updated_at = None
+        ).returning(im)
+        print(f'insert stmt: {insert_stmt}')
+
+        time_now = datetime.utcnow()
+
+        do_update_stmt = insert_stmt.on_conflict_do_update(
+            constraint='unique_instrument_meta',
+            set_=dict(
+                num_channels = self.instrument_meta.num_channels,
+                scan_angle = self.instrument_meta.scan_angle,
+                updated_at = time_now
+            )
+        )
+
+        print(f'do update stmt: {do_update_stmt}')
+
+        try:
+            result = session.execute(do_update_stmt)
+            session.flush()
+            result_row = result.fetchone()
+            action = db_utils.INSERT
+            if result_row.updated_at is not None:
+                action = db_utils.UPDATE
+
+            session.commit()
+            session.close()
+        except Exception as err:
+            message = f'Attempt to INSERT/UPDATE instrument meta record FAILED'
+            error_msg = f'Failed to insert/update record - err: {err}'
+            print(f'error_msg: {error_msg}')
+            session.close()
+        else:
+            message = f'Attempt to {action} instrument meta record SUCCEEDED'
+            error_msg = None
+        
+        results = {}
+        if result_row is not None:
+            results['action'] = action
+            results['data'] = [result_row._mapping]
+            results['id'] = result_row.id
+
+        response = DbActionResponse(
+            self.request_dict,
+            (error_msg is None),
+            message,
+            results,
+            error_msg
+        )
+
+        print(f'response: {response}')
+        return response 
+    
+
+    def get_instrument_metas(self):
+        session = stm.get_session()
+
+        q = session.query(
+            im.id,
+            im.name,
+            im.num_channels,
+            im.scan_angle,
+            im.created_at,
+            im.updated_at
+        ).select_from(
+            im
+        )
+
+        print('Before adding filters to instrument meta request########################')
+        if self.filters is not None and len(self.filters) > 0:
+            for key, value in self.filters.items():
+                q = q.filter(value)
+        
+        print('After adding filters to instrument meta request########################')
```

Comment:
Is this print statement and the one above just for testing/development?

Reply:
yes and no. I use it when I'm trying to understand what's going on with calls from score-monitoring sometimes as a step in the process. probably aren't necessary though, so if you think we should remove these, I think it's fine. in the future, it would be nice to have a flag for verbose print statement options

Reply:
We're still pretty in the weeds development-wise so I think it'll be helpful to keep around. I like the idea of having a verbose flag in the future though, I'll add a low priority task to the backlog for that